hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/qjournal/MiniQJMHACluster.java

```diff
@@ -35,7 +35,7 @@
 import java.util.List;
 import java.util.Random;
 
-public class MiniQJMHACluster {
+public class MiniQJMHACluster implements AutoCloseable {
```

Comment:
Is this change related? I'm not sure whether it should be included in this PR, since it changes the `shutdown` method signature.

Reply:
This doesn't change the call to `shtudown()` at all.  It adds another method called `close()` that allows this class to be `AutoCloseable`, and matches the same pattern used in the `MiniDFSCluster`.

Reply:
Yes, but it changes the signatures so it no longer throw `IOException`, so all the call sites of this method should no longer need to catch that exception. 

I'd prefer to do this separately if not related to this PR.

Reply:
That's was my mistake.  `close()` can't throw an exception, so the try-catch should have appeared in close.  The use of `AutoCloseable` allows try-with-resources which ensures that the resources are released properly.  I've updated `TestRollingUpgrade` because it was consistently failing despite not being related to this change.

I'll fix the exception handling to match the original `shutdown()` signature, and plan on opening a broader test update to take advantage of lessons learned here without other tests.